firestore/firestore_snippets/solution_counters_test.go

```diff
@@ -0,0 +1,56 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"context"
+	"os"
+	"testing"
+
+	"cloud.google.com/go/firestore"
+)
+
+func TestDistributedCounter(t *testing.T) {
+	projectID := os.Getenv("GOLANG_SAMPLES_FIRESTORE_PROJECT")
+	if projectID == "" {
+		t.Skip("Skipping firestore test. Set GOLANG_SAMPLES_FIRESTORE_PROJECT.")
+	}
+
+	ctx := context.Background()
+	client, err := firestore.NewClient(ctx, projectID)
+	if err != nil {
+		t.Fatalf("firestore.NewClient: %v", err)
+	}
+	defer client.Close()
+
+	docRef := client.Collection("counter_samples").Doc("DCounter")
+
+	dc := Counter{3}
+	dc.сreateCounter(ctx, docRef)
+
+	shards := docRef.Collection("shards")
+	docRefs, _ := shards.DocumentRefs(ctx).GetAll()
```

Comment:
Check error. If this fails, it will be nice to know what the error message was.
```suggestion
	docRefs, err := shards.DocumentRefs(ctx).GetAll()
	if err != nil {
		return fmt.Errorf("GetAll: %v", err)
	}
```

Reply:
Done